### PR TITLE
Add REX tool test specs for docs

### PR DESCRIPTION
**Pull request by danieljurek**
@JimSuplizio -- Here are a few packages which fail and one which passes the docs build that can be used to smoke test (but not thoroughly test) the REX tool. 

Since the tool itself doesn't exist today there's no logic to test the tool at this time. This just seeds the test cases so they can be wired up to testing later.

### CODE REVIEW

File: eng/scripts/docs/rex-tool-tests.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
...t.Content.Build.Java2Yaml.ProcessUtility.Execute(String fileName, String commandLineArgs, String cwd, Boolean redirectOutput) in C:\\__w\\1\\s\\src\\java2yaml\\Common\\Utility\\ProcessUtility.cs:line 53\n   at System.Threading.Tasks.Task.Execute()\n--- End of stack trace from previous location where exception was thrown ---\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\n   at Microsoft.Content.Build.Java2Yaml.StepCollection.<RunAsync>d__4.MoveNext() in C:\\__w\\1\\s\\src\\java2yaml\\Steps\\StepCollection.cs:line 41\n--- End of stack trace from previous location where exception was thrown ---\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\n   at Microsoft.Content.Build.Java2Yaml.GenerateDocument.<>c__DisplayClass2_0.<<RunAsync>b__0>d.MoveNext() in C:\\__w\\1\\s\\src\\java2yaml\\Steps\\GenerateDocument.cs:line 13\n--- End of stack trace from previous location where exception was thrown ---\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\n   at Microsoft.Content.Build.Java2Yaml.StepCollection.<RunAsync>d__4.MoveNext() in C:\\__w\\1\\s\\src\\java2yaml\\Steps\\StepCollection.cs:line 28\nFailed in 79482 milliseconds.\n"
```
Comment (weshaggard) [inline eng/scripts/docs/rex-tool-tests.json:21]: Do we really want or need this large comment?

**Review by weshaggard — APPROVED** [review]
One question but otherwise seems good.

File: eng/scripts/docs/rex-tool-tests.json (hunk 1 above)
Comment (danieljurek) [reply eng/scripts/docs/rex-tool-tests.json]: We can remove this test case. Those packages don't publish docs anyway.

**Pull request merged**